force-app/main/default/classes/BDI_ContactService.cls

```diff
@@ -1168,4 +1289,117 @@ public with sharing class BDI_ContactService {
         set;
     }
 
+    /*******************************************************************************************************
+    * @description updates DI records with salesforce ids if it finds a match using duplicate management rules
+    */
+    private void performDuplicateManagement() {
+        Map<SObject,DataImport__c[]> c1SObjectToDIs = new Map<SObject,DataImport__c[]>();
+        Map<SObject,DataImport__c[]> c2SObjectToDIs = new Map<SObject,DataImport__c[]>();
+
+        for (DataImport__c di : bdi.listDI) {
+            // skip DIs that already have an error
+            if (di.Status__c == BDI_DataImport_API.bdiFailed)
+                continue;
+
+            // Only add a contact for matching if there is information for it and there isn't already an Id specified
+            if (di.Contact1Imported__c == null && isContactSpecified(di, 'Contact1')) {
+                addSobjToDIMap(c1SObjectToDIs, di, 1);
+            }
+
+            // Only add a contact for matching if there is information for it and there isn't already an Id specified
+            if (di.Contact2Imported__c == null && isContactSpecified(di, 'Contact2')) {
+                addSobjToDIMap(c2SObjectToDIs, di, 2);
+            }
+        }
+
+        // Return if there were no sObjects added for dupe matching.
+        if (sObjsForDuplicateManagement == null || sObjsForDuplicateManagement.size() == 0) {
+            return;
+        }   
+
+        sObjToMatchDetails = dupeMgmtUtil.findDuplicatesBulk(new List<SObject>(sObjsForDuplicateManagement));
+
+        // Return if there were no dupe results
+        if (sObjToMatchDetails == null || sObjToMatchDetails.size() == 0) {
```

Comment:
`sObjToMatchDetails` is never going to be `null` because it is newed up in `UTIL_DuplicateMgmt`